common/test/java/com/couchbase/lite/logging/LogTest.kt

```diff
@@ -548,61 +549,102 @@ class LogTest : BaseDbTest() {
     // There was a bug that caused an attempt to log a message from within the
     // LiteCore logging callback to hang the callback thread
     //
-    // If there is a file logger, an attempt to log a message from within the
-    // callback will forward the log message to LiteCore.  That, in turn, will
-    // call c4log_getDomain.  If c4log_getDomain attempts to seize the same
-    // lock that is already held by the LiteCore thread, the thread will hang
+    // If there is a file logger, when the platform logs a message it will forwarded
+    // to LiteCore.  That, in turn, will call c4log_getDomain.  If c4log_getDomain
+    // attempts to re-acquire the same non-reentrant lock that the thread already holds,
+    // it will deadlock
+    //
+    // Since CBL-6800, this test must test only that the platfrom cannot generate recursive
+    // log from the LiteCore callback thread.  A an attempt to log from some other thread
+    // during the callback, will be blocked until the callback returns but cannot, directly,
+    // deadlock.
     @Test
     fun testRecursiveLogging() {
         val logSinks = assertNonNull(LogSinksImpl.getLogSinks())
         logSinks.file = FileLogSink.Builder().setDirectory(scratchDirPath!!).build()
 
-        testInCallback { Log.e(LogDomain.DATABASE, "BANG!!!") }
+        val startLatch = CountDownLatch(1)
+        val finishLatch = CountDownLatch(1)
+
+        var success: Boolean? = null
+
+        assertNonNull(logSinks.c4Log).setCallbackInstrumentation(object : C4Log.Instrumentation {
+            override fun onLogToCore(domain: LogDomain, level: LogLevel, message: String): Boolean {
+                success = false
+                return false // do not allow a recursive call!!  It will deadlock!
+            }
+
+            override fun onCallback(c4Domain: String?, c4Level: Int, message: String?): Boolean {
+                startLatch.countDown()
+
+                // Try to generate a recursive call to LiteCore logging
+                Log.e(LogDomain.DATABASE, "BANG!!!")
```

Comment:
Probably this question is rephrasing what is described as the comment of this test but just want to make sure I understand it correctly.

Is it correct that this test tests that Log.e() will not cause `logToCore()` to be called as if that happens, it will cause a deadlock from `c4log_getDomain` call?